### PR TITLE
custom: replace vexWiggleArpeggioUp by wiggleArpeggiatoUp

### LINKED ISSUE #1560: Custom glyphs: vexWiggleArpeggioUp
vexWiggleArpeggioUp is not a SMUFL standard glyph. A rotated wiggleArpeggiatoUp should be used instead.

### PR DESCRIPTION
**Pull request by rvilarl**
fixes #1560

### CODE REVIEW

**Comment by rvilarl** [comment]
Visual differences (current first):
Bravura
pptr-StringNumber Multi_Voice_With_Strokes__String___Finger_Numbers Bravura svg_current
pptr-StringNumber Multi_Voice_With_Strokes__String___Finger_Numbers Bravura svg_reference
Gonville
pptr-StringNumber Multi_Voice_With_Strokes__String___Finger_Numbers Gonville svg_current
pptr-StringNumber Multi_Voice_With_Strokes__String___Finger_Numbers Gonville svg_reference
Leland
pptr-StringNumber Multi_Voice_With_Strokes__String___Finger_Numbers Leland svg_current
pptr-StringNumber Multi_Voice_With_Strokes__String___Finger_Numbers Leland svg_reference
Petaluma
pptr-StringNumber Multi_Voice_With_Strokes__String___Finger_Numbers Petaluma svg_current
pptr-StringNumber Multi_Voice_With_Strokes__String___Finger_Numbers Petaluma svg_reference

**Comment by sschmidTU** [comment]
I liked the finer lines of the previous version, but it's not a big deal.
Any way to make the line less bold? (if it's too much work, don't worry about it)

**Comment by rvilarl** [comment]
Thanks a lot for the feedback! I will generate a new GonvilleSmufl including the arpegiattos (I believe that those from Gonville were  used to generate the custom  glyphs) and figure out how to align the arrows (this will be much easier on vexflow5)

**Comment by rvilarl** [comment]
@mscuthbert @sschmidTU I have updated GonvilleSmufl to include its original finer lines, the other fonts define their own and I think it makes sense to leave the creator's choice. I have also fix the miscentered arrow heads. @mscuthbert in regards to the R, we should use PetalumaScript to fix that. Current first.

Bravura
pptr-StringNumber Multi_Voice_With_Strokes__String___Finger_Numbers Bravura svg_current
pptr-StringNumber Multi_Voice_With_Strokes__String___Finger_Numbers Bravura svg_reference
Gonville
pptr-StringNumber Multi_Voice_With_Strokes__String___Finger_Numbers Gonville svg_current
pptr-StringNumber Multi_Voice_With_Strokes__String___Finger_Numbers Gonville svg_reference
Leland
pptr-StringNumber Multi_Voice_With_Strokes__String___Finger_Numbers Leland svg_current
pptr-StringNumber Multi_Voice_With_Strokes__String___Finger_Numbers Leland svg_reference
Petaluma
pptr-StringNumber Multi_Voice_With_Strokes__String___Finger_Numbers Petaluma svg_current
pptr-StringNumber Multi_Voice_With_Strokes__String___Finger_Numbers Petaluma svg_reference

**Comment by ronyeh** [comment]
The down arrows still look not centered (at least with Leland).

I agree that finer lines looks better. Otherwise it adds a lot of visual weight, and doesn't match as well with the stems. Although I'd have to see real life examples to see whether this symbol should be thicker than the stems, or the same.

Is there any other glyph that can be used for this purpose?

Or should we programmatically draw our own splines for this?

**Comment by rvilarl** [comment]
> Is there any other glyph that can be used for this purpose?

see https://w3c.github.io/smufl/latest/tables/multi-segment-lines.html

> Or should we programmatically draw our own splines for this?

I would not do that. We could create a custom.otf but I think it is better to use the original font.

File: src/strokes.ts [hunk 1]
```diff
@@ -207,36 +179,65 @@ export class Stroke extends Modifier {
...
+          topY - (h * 3) / 4,
+          this.render_options.font_scale,
+          arrow == 'arrowheadBlackDown' ? 'wiggleArpeggiatoDown' : 'wiggleArpeggiatoUp'
+        );
+      }
+      if (arrow == 'arrowheadBlackDown') {
+        // Draw the arrow head
+        Glyph.renderGlyph(
+          ctx,
+          x + this.x_shift - h / 4 + i - 10 + 4,
```
Comment (ronyeh) [inline src/strokes.ts:226]: Can you explain some of these calculations, and also in some of the blocks above as well? Someone looking at this code in the future won't be able to know exactly what we are trying to accomplish here.

I just want to make sure the arrow heads are centered horizontally in all of our fonts. Right now some of them don't look centered to me. 

Can you generate zoomed in images with the wiggle + arrowhead for all fonts? We need to position the arrow head so it looks centered (and touching the wiggle at the narrowest point).

Comment (rvilarl) [reply src/strokes.ts]: Code reworked to make it easier to understand and easier to comment

**Comment by rvilarl** [comment]
After the rework (current first):
Bravura
pptr-StringNumber Multi_Voice_With_Strokes__String___Finger_Numbers Bravura svg_current
pptr-StringNumber Multi_Voice_With_Strokes__String___Finger_Numbers Bravura svg_reference
Gonville
pptr-StringNumber Multi_Voice_With_Strokes__String___Finger_Numbers Gonville svg_current
pptr-StringNumber Multi_Voice_With_Strokes__String___Finger_Numbers Gonville svg_reference
Leland
pptr-StringNumber Multi_Voice_With_Strokes__String___Finger_Numbers Leland svg_current
pptr-StringNumber Multi_Voice_With_Strokes__String___Finger_Numbers Leland svg_reference
Petaluma
pptr-StringNumber Multi_Voice_With_Strokes__String___Finger_Numbers Petaluma svg_current
pptr-StringNumber Multi_Voice_With_Strokes__String___Finger_Numbers Petaluma svg_reference

**Comment by ronyeh** [comment]
Hi!

Can we remove docs/api from this PR? You can submit a separate PR for docs once we merge all outstanding PRs.

Two things:
- In each of the font samples you provided, the arrow heads are not centered relative to the straight lines. So I think in adjusting the arrow heads for the squiggles, we have made them off-center relative to the straight lines (on the right side of the test cases). I know some of them weren't perfectly centered before, but it has gotten a little worse. Might as well take this opportunity to clean it up if possible.
- I would prefer if the arrow head attaches at the wiggle's narrowest point (or most centered point). The arrowhead should look like an arrow, and when it attaches at the thickest part of the wiggle, it doesn't look as much like an arrow (because the wings of the arrowhead are too similar to the thickness of the wiggle). See the Petaluma image, for example. It would be nice if we can figure out a font-indepent way to attach the arrowhead to the wiggle.

**Comment by rvilarl** [comment]
Let us leave that for 5.0

**Comment by ronyeh** [comment]
When we revisit this in 5.0, we should consider using: 

`wiggleArpeggiatoUpArrow` and `wiggleArpeggiatoDownArrow`

Those include the arrowheads and are centered correctly.

**Comment by rvilarl** [comment]
> wiggleArpeggiatoUpArrow

I considered them but they have a fix length. In 5.0 it also easier to center them using the fonts directly.

**Comment by ronyeh** [comment]
For future discussion, this is how Dorico handles it:
https://steinberg.help/dorico/v2/en/dorico/topics/notation_reference/notation_reference_arpeggio_signs_c.html

https://w3c.github.io/smufl/latest/tables/multi-segment-lines.html

image

**Pull request closed without merging**